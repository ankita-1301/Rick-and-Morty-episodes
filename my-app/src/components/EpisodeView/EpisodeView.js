import React from "react";
import "../../styles.css";
import "antd/dist/antd.css";
import { Card, Avatar } from "antd";
import { UserOutlined } from "@ant-design/icons";
import { Link } from "react-router-dom";

//Component for rendering single episode
export default class EpisodeView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      characters: [],
    };
  }

  componentDidMount() {
    this.fetchCharacters();
  }

  //fetching all the characters that the episode contains
  fetchCharacters = () => {
    const {
      location: {
        state: { data },
      },
    } = this.props;
    data.characters.forEach((characterPage) => {
      fetch(`${characterPage}`)
        .then((response) => response.json())
        .then((eachCharacter) => {
          this.setState({
            characters: [...this.state.characters, ...[eachCharacter]],
          });
        })
        .catch((err) => console.log(err));
    });
  };

  render() {
    const { characters } = this.state;
    const {
      location: {
        state: { data },
      },
    } = this.props;

    const gridStyle = {
      width: "25%",
      textAlign: "center",
      padding: "20px",
    };

    return (
      <Card
        title={`${data.episode}â€™s characters`}
        className="character-view-card"
      >
        {characters.map((char) => {
          return (
            <Link
              to={{
                pathname: `/characters/character/${char.id}`,
                state: { data: char },
              }}
              key={char.id}
            >
              <Card.Grid style={gridStyle} key={char.id}>
                <Avatar
                  shape="square"
                  size={100}
                  icon={<UserOutlined />}
                  src={char.image}
                />
                <h3>{char.name}</h3>
              </Card.Grid>
            </Link>
          );
        })}
      </Card>
    );
  }
}
