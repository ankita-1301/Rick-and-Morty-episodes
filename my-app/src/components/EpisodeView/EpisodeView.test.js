import React from "react";
import Enzyme, { shallow } from "enzyme";
import EpisodeView from "./EpisodeView";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

//test setup for episodeView component

const state = {
  state: {
    data: {
      id: 2,
      name: "Lawnmower Dog",
      air_date: "December 9, 2013",
      episode: "S01E02",
      characters: [
        "https://rickandmortyapi.com/api/character/1",
        "https://rickandmortyapi.com/api/character/2",
      ],
      url: "https://rickandmortyapi.com/api/episode/2",
      created: "2017-11-10T12:56:33.916Z",
    },
  },
};

const props = {};

test("EpisodeView component snapshot", () => {
  const episodeView = renderer.create(
    <EpisodeView {...props} location={state} />
  );
  let epiTree = episodeView.toJSON();
  expect(epiTree).toMatchSnapshot();
});

describe("Test episode view component", () => {
  describe("Has rendered with mock props", () => {
    let component;
    beforeEach(() => {
      component = shallow(<EpisodeView {...props} location={state} />);
    });

    it("Renders episode view without error", () => {
      expect(component.length).toBe(1);
    });
  });
});
