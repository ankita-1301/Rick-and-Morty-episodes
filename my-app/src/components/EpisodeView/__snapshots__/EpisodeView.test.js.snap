// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes component snapshot 1`] = `
<div
  className="ant-card character-view-card ant-card-bordered"
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        S01E02â€™s characters
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  />
</div>
`;
